src/GameLogic/Entities/Iceblock.ts

```diff
@@ -54,18 +58,25 @@ export class Iceblock implements Entity {
 		}
 
 		if (this.melting) {
-			this.containedEntity.x = this.x;
-			this.containedEntity.y = this.y;
-			this.containedEntity.prevX = this.x;
-			this.containedEntity.prevY = this.y;
-
-			level.entities.addEntity(this.containedEntity);
+			if (this.containedEntity !== undefined) {
+				this.containedEntity.x = this.x;
+				this.containedEntity.y = this.y;
+				this.containedEntity.prevX = this.x;
+				this.containedEntity.prevY = this.y;
+
+				level.entities.addEntity(this.containedEntity);
+			}
 			turnState.killEntity(this, TurnEventType.EntityKilled);
 		}
 	}
 
 	public clone(): Iceblock {
-		const clone = new Iceblock(this.containedEntity.clone());
+		let clone;
+		if (this.containedEntity === undefined) {
+			clone = new Iceblock(undefined);
+		} else {
+			clone = new Iceblock(this.containedEntity.clone());
```

Comment:
I think the whole `if` can be rewritten with `new Iceblock(this.containedEntity?.clone())`